### ISSUE TITLE
Need to recognize Kindle highlights URL that it takes you to after logging in.

### ISSUE THREAD

**Issue by GitMurf** [GitMurf/roam-highlighter#63]
This stemmed from Issue #61 

If you are not currently logged in to Amazon and it asks you to login when you go to https://read.amazon.com/notebook then it appears it takes you to a bit of a different URL which Roam highlighter does not recognize and therefore doesn't show the Kindle extract options / buttons.

The solution is to simply click the link again in Roam Highlighter settings (once logged in to Amazon) or simply paste URL https://read.amazon.com/notebook in browser again, and then you should be good to go! See screenshot and GIF below for more details.

But what this Issue is for is to update Highlighter to recognize that other URL that Amazon brings you to when you login. See this GIF for example of the issue.

![Upm7XymT8p](https://user-images.githubusercontent.com/64155612/93898014-981d0c80-fca7-11ea-9c65-bc88eba67d94.gif)

**Comment by GitMurf** [comment]
From #72 

When I go to the settings page and klick on Kindle Notes & Highlights Amazon shows me my Highlights correctly.
But when I chose "get Kindle highlights" I get a messagebox telling me:
Not on kindle page. Go to: https://read.amazon.com/notebook

The page I'm on is https://read.amazon.com/notebook
Source of the issue might be that I'm on the German Amazon account, so maybe the code to recognize if I'm on the Notebook page is checking some elements that are localized?

Would be great if this could be fixed, as I'm highly interested in using it for copying my notes into markdown.

**Comment by GitMurf** [comment]
From #61 

It seems there is an issue for non-US amazon accounts though. For instance in France, the URL is https://lire.amazon.fr/notebook ("lire" means "read", you get it)
I don't see the Kindle Highlights button, I guess that it is because that URL is not recognized.
Unfortunately I don't have access to my books if I log to read.amazon.com (probably to book copyright restrictions in other countries)

Any hint on how to solve it?

**Comment by GitMurf** [comment]
> Browse to read.amazon.com.
Click "Show notes and marks" button in the Kindle web toolbar.
Kindle web page displays book highlights in a simulated modal dialog (KindleWebReader.pdf), unlike the way it appears in the Roam Highlighter demo.
Open Roam Highlighter Settings.
Kindle Highlights button does not appear in the Settings pane (RoamHighlighterSettings.pdf).

If this issue occurs, it should be resolved by clicking the link again in the Roam-highlighter settings. OR you can just enter directly into the browser address: `https://read.amazon.com`_**/notebook**_

The key is adding the **_/notebook_** on the end.

**Comment by GitMurf** [comment]
@lwrogersjr your issue from #61 should be completely resolved now. See above instructions if it happens again.

**Comment by GitMurf** [comment]
@mattwien and @clement-moulin-frier

I have just made some tweaks to hopefully get closer to working for France / Germany. It may still not work YET, but at least it should give me the info I need to fix it completely. You should both be able to go to your respective read.amazon.com/notebook pages and select the button to extract the highlights. If you get a pop-up box with an error/warning, please take a screenshot of it or paste the text in here for me to review. That will be the last piece I need to be able to get it working for you.

Note you will need to wait a day or two for my updates to get approved by Google and the auto update pushed out. When it loads and says v1.9.8 you will know it has updated.

Thanks!
-Shawn

**Comment by mattwien** [comment]
@GitMurf , thank you for the update. I'll check it as soon as the update is available and let you know of the results. 
Also let me know if you need any further info e.g. about the page strucutre that we have here on the German Amazon accounts.

**Comment by GitMurf** [comment]
@mattwien thanks actually if you could get me a screenshot showing the exact URL that German account leads to along with hovering over the tab to show what it uses as the page title. I may be able to fix things before this new release as opposed to just waiting for more info after the release ;-)

![image](https://user-images.githubusercontent.com/64155612/96921867-f33b4e00-1463-11eb-8176-23b136d8aea9.png)

Also can you do a right-click > Inspect to show the HTML of the page so I can see if any of the classes/IDs are spelled/named differently due to language differences? First screenshot is what it looks like on the page and then the following are of the different HTML elements that are important.

![image](https://user-images.githubusercontent.com/64155612/96922505-cb98b580-1464-11eb-9f55-990f94a31538.png)

![image](https://user-images.githubusercontent.com/64155612/96922190-5fb64d00-1464-11eb-8b2a-3429013f7041.png)

![image](https://user-images.githubusercontent.com/64155612/96922352-94c29f80-1464-11eb-80a7-46aae5d19830.png)

![image](https://user-images.githubusercontent.com/64155612/96922430-b1f76e00-1464-11eb-803c-4d44ac190d81.png)

**Comment by mattwien** [comment]
@GitMurf :
Thanks for the explanation. here the information you need:
Englisch Word Note is translated to Notiz (as you can read in the screenshots).

**Links:**
There are basically multiple possible links.
When I’m on the Kindle page and select the Notes I get to:
https://lesen.amazon.de/notebook?ref_=kcr_notebook_lib
but https://read.amazon.com/notebook?ref_=kcr_notebook_lib works as well.

**Page title is:**
![image](https://user-images.githubusercontent.com/70956195/97039248-66c88280-156c-11eb-8c85-5d89773bc6d2.png)

**Header section**
![image](https://user-images.githubusercontent.com/70956195/97039265-6f20bd80-156c-11eb-8f4d-a25ccc26f8fc.png)

**Notes Header:**
![image](https://user-images.githubusercontent.com/70956195/97039279-7647cb80-156c-11eb-9f9d-f26e50fbef37.png)


Thank you again for your support!

**Comment by GitMurf** [comment]
> Thanks for the explanation. here the information you need:

@mattwien wow! thanks so much for the great info! It isn't often I can get folks to provide the level of detail to make my job easy ;-) Appreciate it! I will make some further changes before the new release that will give a better chance at success for us!

@mattwien Would you be interested in me showing you how to actually load my "DEV" version of the extension? That way you can test it out before I submit release to Chrome for their approval and then update push? That takes a few days usually. Let me know! thanks.

**Comment by mattwien** [comment]
Thank you @GitMurf for the feedback :)
Yes, would be interested to test the DEV version - would help us to accelerate the process for both of us ;-)

**Comment by GitMurf** [comment]
@mattwien ￼here are the steps:

1) download these files and put into a folder named DEV anywhere on your computer.

2) open up chrome and go to your extensions by entering this in browser URL: `chrome://extensions/`

3) in top right corner there should be a toggle for Developer Mode. Turn it on.

4) disable current roam-highlighter extension.

5) click the button in upper left that says "Load Unpacked".

6) select/open the DEV folder from step 1.

Now you should have the roam-highlighter dev extension loaded and when you run it (click the highlighter icon) you should see version 1.9.8 when it loads.

Let me know if you get it working and then we can start testing things with the kindle notes.

**Comment by mattwien** [comment]
I'm quite new to GIT but am able to clone, commit, pull and push in/to repositories 😀
Thanks for the explanation, I was able to install the DEV version and now I get the requried content as long as I'm on the English link / klick in the roam-highlighter on "kindle notes&highlights" button. If I'm on the german link it's not showing the relevant area to save capture the highlights yet - but that's not a problem for me.


![image](https://user-images.githubusercontent.com/70956195/97108303-ed947100-16cc-11eb-9f2a-f04b5acab3c7.png)

Here the Link, as it's captures from the tool, that is currently not yet showing the respective features in the tool:
[Kindle: Ihre Notizen und Markierungen](https://lesen.amazon.de/notebook?ref_=kcr_notebook_lib) [[Obsidian-Highlights]]

**Comment by GitMurf** [comment]
@mattwien ok I just made an update. Can you see if the German link works now? Thanks! If you cloned the Repo then simply PULL the update down. If you just downloaded the .zip previously you can simply download the updated `highlighter.js` file from here [(highlighter.js)](https://github.com/GitMurf/roam-highlighter/blob/master/dev/highlighter.js) and replace it in the DEV folder and then do the "Load Unpacked" again, refresh your browser page and try it.

To download the `highlighter.js` file, click the link above and then right-click "Raw" and select "Save link as..." and it should save it as a .js Javascript file that you can replace in your DEV folder.

![image](https://user-images.githubusercontent.com/64155612/97113542-55849080-16a8-11eb-80d0-8af0642201ff.png)

**Comment by mattwien** [comment]
Pulled the repo, resinstalled the extension in Chrome and voilá: It works. 

Thanky you very much for the quick update!

**Comment by GitMurf** [comment]
> Pulled the repo, resinstalled the extension in Chrome and voilá: It works.
> 
> Thanky you very much for the quick update!

@mattwien Woot! Awesome. Thank you for working with me on this! :) So just to clarify, now it works for BOTH the US/English amazon link and the German amazon link? All the highlights and metadata is pulled fine for both?

**Comment by mattwien** [comment]
@GitMurf yes. All working find. I double checked some aspects, personal notes (I rarely use them on my Kindle - but do them directly in my notes system) are there, links, title, autor,... Looks very complete.
The only issue I have left is that I'm using a different system to handle my notes, and would like to have some slightly different formatting - but as it's also using markdown I'm very happy with the format I get when activating the Obsidian style.
